design/marketplace.md

```diff
@@ -0,0 +1,279 @@
+                                          ---- fill slot --- | Host |
+                                                             --------
+
+
+The storage contract contains the content identifier, so that hosts can locate
+and download the content. It also contains the reward that hosts receive for
+storing the data and the collateral that hosts are expected to deposit. It
+contains parameters pertaining to storage proofs and erasure coding. And
+finally, it contains the amount of hosts that are expected to store the content,
+including a small amount of host losses that can be tolerated.
+
+
+    StorageContract
+
+      cid                             # content identifier
+
+      reward                          # tokens payed per second per filled slot
+      collateral                      # amount of collateral required per host
+
+      proof probability               # frequency at which proofs are required
+      proof parameters                # proof of retrievability parameters
+      erasure coding                  # erasure coding parameters
+      dispersal                       # dispersal parameter
+
+      hosts                           # amount of storage hosts (including loss)
+      loss                            # number of allowed host losses
+
+      slots                           # assigned host slots
+
+      timeout                         # slots need to be filled before timeout
+
+Slots
+-----
+
+Initially all host slots are empty. An empty slot can be filled by anyone by
+submitting a correct storage proof together with collateral.
+
+
+        collateral                              collateral
+          proof    proof          missed          proof                missed
+            |        |              |               |                    |
+            v        v              v               v                    v
+            -------------------------------------------------------------------
+     slot:  |///////////////////////|               |////////////////////|
+            -------------------------------------------------------------------
+                                    |                                    |
+                                    v                                    v
+                                collateral                           collateral
+                                  burned                               burned
+
+
+
+            ---------------- time ---------------->
+
+
+The time interval that a slot is filled by a host determines the host payout;
+for every second of the interval a certain amount of tokens are awarded to the
+host. Hosts that fill a slot are required to submit frequent proofs of storage.
+
+When a proof is missed, the collateral associated with a slot is used to pay a
+fee to the one who marked the proof as missing. The rest of the slot collateral
+is reserved for repairs. The slot is now considered empty again until another
```

Comment:
This feels a bit harsh. Maybe we could allow for a few missed proofs before the the host loses their collateral and the slot is freed up?

Reply:
Yes, I agree. I will change the wording here to indicate that a number of proofs need to be missing before triggering repair.